src/main/java/fi/aalto/cs/apluscourses/intellij/model/ScalaSdk.java

```diff
@@ -80,24 +78,28 @@ public String getFileName() {
 
   @Override
   protected String[] getUris() {
-    return getUris(CLASSES);
+    return getUris(CLASSES, path -> VfsUtil.getUrlForLibraryRoot(path.toFile()));
```

Comment:
I don't really like to use platform's implementation for something that should just convert `file:///foo/bar/baz.jar` to `jar:///foo/bar/baz.jar!/`. The thing is, this format of JAR URI that is used by IntelliJ IDEA is _non-standard_ (there should be the whole URL, including the scheme, after `jar:`) and IDEA doesn't support standard JAR URLs (For standard, see [these](https://www.iana.org/assignments/uri-schemes/prov/jar) [sources](http://java.sun.com/javase/6/docs/api/java/net/JarURLConnection.html) referenced by [Wikipedia](https://en.wikipedia.org/wiki/List_of_URI_schemes)).

As I'm not aware of any existing specs for this non-standard URL version, it is not meaningful to write an own implementation based on vague guess of how it's probably meant to work. Fortunately, there is this utility function available in IntelliJ's OpenAPI, so we can use it and assume it's compatible with whatever is the "IntelliJ's way" here.